pkg/apiserver/rest/usecase/webhook.go

```diff
@@ -70,42 +74,126 @@ func (c *webhookUsecaseImpl) HandleApplicationWebhook(ctx context.Context, token
+		component := &model.ApplicationComponent{
+			AppPrimaryKey: webhookTrigger.AppPrimaryKey,
+			Name:          comp,
 		}
-		for comp, properties := range webhookReq.Upgrade {
-			component := &model.ApplicationComponent{
-				AppPrimaryKey: webhookTrigger.AppPrimaryKey,
-				Name:          comp,
-			}
-			if err := c.ds.Get(ctx, component); err != nil {
-				if errors.Is(err, datastore.ErrRecordNotExist) {
-					return nil, bcode.ErrApplicationComponetNotExist
-				}
-				return nil, err
-			}
-			merge, err := envbinding.MergeRawExtension(component.Properties.RawExtension(), properties.RawExtension())
-			if err != nil {
-				return nil, err
-			}
-			prop, err := model.NewJSONStructByStruct(merge)
-			if err != nil {
-				return nil, err
-			}
-			component.Properties = prop
-			if err := c.ds.Put(ctx, component); err != nil {
-				return nil, err
+		if err := c.ds.Get(ctx, component); err != nil {
+			if errors.Is(err, datastore.ErrRecordNotExist) {
+				return nil, bcode.ErrApplicationComponetNotExist
 			}
+			return nil, err
 		}
-		return c.applicationUsecase.Deploy(ctx, app, apisv1.ApplicationDeployRequest{
-			WorkflowName: webhookTrigger.WorkflowName,
-			Note:         "triggered by webhook",
-			TriggerType:  apisv1.TriggerTypeWebhook,
-			Force:        true,
-			CodeInfo:     webhookReq.CodeInfo,
-		})
-	default:
-		return nil, bcode.ErrInvalidWebhookPayloadType
+		merge, err := envbinding.MergeRawExtension(component.Properties.RawExtension(), properties.RawExtension())
+		if err != nil {
+			return nil, err
+		}
+		prop, err := model.NewJSONStructByStruct(merge)
+		if err != nil {
+			return nil, err
+		}
+		component.Properties = prop
+		if err := c.ds.Put(ctx, component); err != nil {
+			return nil, err
+		}
+	}
+	return c.applicationUsecase.Deploy(ctx, app, apisv1.ApplicationDeployRequest{
+		WorkflowName: webhookTrigger.WorkflowName,
+		Note:         "triggered by webhook custom",
+		TriggerType:  apisv1.TriggerTypeWebhook,
+		Force:        true,
+		CodeInfo:     webhookReq.CodeInfo,
+	})
+}
+
+func (c *webhookUsecaseImpl) handleACRWebhook(ctx context.Context, req *restful.Request, webhookTrigger *model.ApplicationTrigger, app *model.Application) (*apisv1.ApplicationDeployResponse, error) {
+	var acrReq apisv1.HandleApplicationTriggerACRRequest
+	if err := req.ReadEntity(&acrReq); err != nil {
+		return nil, bcode.ErrInvalidWebhookPayloadBody
+	}
+	if webhookTrigger.ComponentName == "" {
+		return nil, bcode.ErrApplicationComponetNotExist
+	}
+	component := &model.ApplicationComponent{
+		AppPrimaryKey: webhookTrigger.AppPrimaryKey,
+		Name:          webhookTrigger.ComponentName,
+	}
+	if err := c.ds.Get(ctx, component); err != nil {
+		if errors.Is(err, datastore.ErrRecordNotExist) {
+			return nil, bcode.ErrApplicationComponetNotExist
+		}
+		return nil, err
+	}
+
+	image := fmt.Sprintf("registry.%s.aliyuncs.com/%s:%s", acrReq.Repository.Region, acrReq.Repository.RepoFullName, acrReq.PushData.Tag)
```

Comment:
Does the request body have not a full name?

Reply:
No.

The request body is like:

```json
{
    "push_data": {
        "digest": "sha256:457f4aa83fc9a6663ab9d1b0a6e2dce25a12a943ed5bf2c1747c58d48bbb4917", 
        "pushed_at": "2016-11-29 12:25:46", 
        "tag": "latest"
    }, 
    "repository": {
        "date_created": "2016-10-28 21:31:42", 
        "name": "repoTest", 
        "namespace": "namespace", 
        "region": "cn-hangzhou", 
        "repo_authentication_type": "NO_CERTIFIED", 
        "repo_full_name": "namespace/repoTest", 
        "repo_origin_type": "NO_CERTIFIED", 
        "repo_type": "PUBLIC"
    }
}
```